tiledb/sm/stats/global_stats.h

```diff
@@ -135,14 +135,22 @@ class GlobalStats {
   mutable std::mutex mtx_;
 
   /** The aggregated stats. */
-  std::list<shared_ptr<stats::Stats>> registered_stats_;
+  std::list<std::weak_ptr<stats::Stats>> registered_stats_;
 
   /* ****************************** */
   /*       PRIVATE FUNCTIONS        */
   /* ****************************** */
 
   /** Dump the current registered stats. */
   std::string dump_registered_stats() const;
+
+  /** iterate over raw stats calling f() */
+  template <class FuncT>
+  void iterate(const FuncT&);
+
+  /** iterate over raw stats calling f() */
+  template <class FuncT>
+  void iterate(const FuncT&) const;
```

Comment:
The documentation point about const-qualification should likely to applied here at the declaration. The distinction in the iterations is worth highlighting.